Sources/PackageModel/IdentityResolver.swift

```diff
@@ -20,6 +21,10 @@ public protocol IdentityResolver {
     func resolveIdentity(for path: AbsolutePath) throws -> PackageIdentity
     func mappedLocation(for location: String) -> String
     func mappedIdentity(for identity: PackageIdentity) throws -> PackageIdentity
+
+    func mappedDependency(for dependency: PackageDependency, fileSystem: FileSystem) throws -> PackageDependency
```

Comment:
nit: missing doc comment

Reply:
Same here, I don't think I can describe the logic here.